sql/core/src/main/java/org/apache/spark/sql/sources/v2/SupportsBatchRead.java

```diff
@@ -24,10 +24,10 @@
 /**
  * An empty mix-in interface for {@link Table}, to indicate this table supports batch scan.
  * <p>
- * If a {@link Table} implements this interface, its {@link Table#newScanBuilder(DataSourceOptions)}
- * must return a {@link ScanBuilder} that builds {@link Scan} with {@link Scan#toBatch()}
- * implemented.
+ * If a {@link Table} implements this interface, the
+ * {@link SupportsRead#newScanBuilder(DataSourceOptions)} must return a {@link ScanBuilder} that
+ * builds {@link Scan} with {@link Scan#toBatch()} implemented.
  * </p>
  */
 @Evolving
-public interface SupportsBatchRead extends Table { }
+public interface SupportsBatchRead extends SupportsRead { }
```

Comment:
@cloud-fan  How about the following?
```scala
-public interface SupportsBatchRead extends SupportsRead { }
+public interface SupportsBatchRead extends Table, SupportsRead { }
```
```scala
-interface SupportsRead extends Table {
+interface SupportsRead {
```

Reply:
`SupportsRead` only makes sense under the table context. It also simplifies the code
```
public interface SupportsBatchRead extends Table, SupportsRead { }
public interface SupportsMicroBatchRead extends Table, SupportsRead { }
public interface SupportsContinuousRead extends Table, SupportsRead { }
```
to
```
public interface SupportsBatchRead extends SupportsRead { }
public interface SupportsMicroBatchRead extends SupportsRead { }
public interface SupportsContinuousRead extends SupportsRead { }
```

BTW `SupportsRead` is package private.